### PR TITLE
Adds is_public field to wishlist and also enables public wishlists

### PR DESCRIPTION
**Pull request by riteshshrv**

### CODE REVIEW

File: tests/test_wishlist.py [hunk 1]
```diff
@@ -648,7 +659,55 @@ def test_0050_render_single_wishlist(self):
...
+                user2_private_wishlist, = Wishlist.create([{
+                    'nereid_user': self.registered_user2,
+                    'name': 'Books I don\'t Want to Read!',
+                }])
+
+                user3_public_wishlist, = Wishlist.create([{
+                    'nereid_user': self.registered_user3,
+                    'name': 'Books I don\'t like!',
+                    'is_public': True
+                }])
```
Comment (priyankarani) [inline tests/test_wishlist.py]: create wishlist outside test client

File: tests/test_wishlist.py [hunk 2]
```diff
@@ -648,7 +659,55 @@ def test_0050_render_single_wishlist(self):
...
+                rv = c.get(
+                    '/wishlists/%d' % (user2_private_wishlist, )
+                )
+                self.assertEqual(rv.status_code, 404)
+
+                # User trying to access public wishlist of another user
+                rv = c.get(
+                    '/wishlists/%d' % (user3_public_wishlist, )
+                )
+                self.assertEqual(rv.status_code, 200)
```
Comment (priyankarani) [inline tests/test_wishlist.py]: good job :)

File: tests/test_wishlist.py [hunk 3]
```diff
@@ -648,7 +659,55 @@ def test_0050_render_single_wishlist(self):
...
+
+    def test_0070_rename_wishlist(self):
```
Comment (priyankarani) [inline tests/test_wishlist.py]: why this change ?

File: tests/test_wishlist.py [hunk 4]
```diff
@@ -701,7 +760,7 @@ def test_0060_rename_wishlist(self):
                 wishlist = Wishlist(wishlist.id)    # reload the record
                 self.assertEqual(wishlist.name, 'Test2')
 
-    def test_0070_copy_product(self):
+    def test_0080_copy_product(self):
```
Comment (priyankarani) [inline tests/test_wishlist.py]: why this change ?

File: tests/test_wishlist.py [hunk 5]
```diff
@@ -648,7 +659,55 @@ def test_0050_render_single_wishlist(self):
...
+                    '/wishlists/%d' % (user2_private_wishlist, )
+                )
+                self.assertEqual(rv.status_code, 404)
+
+                # User trying to access public wishlist of another user
+                rv = c.get(
+                    '/wishlists/%d' % (user3_public_wishlist, )
+                )
+                self.assertEqual(rv.status_code, 200)
+
```
Comment (priyankarani) [inline tests/test_wishlist.py]: also check for wishlist as guest user

File: tests/test_wishlist.py (hunk 3 above)
Comment (riteshshrv) [inline tests/test_wishlist.py]: its the change in comments and test number orders

File: tests/test_wishlist.py (hunk 4 above)
Comment (riteshshrv) [inline tests/test_wishlist.py]: because test_public_private_wishlist has number 0060 and it is placed after render_single_wishlist with number 0050, so i had to adjust (increment) test numbers for all of them below test_0050
